test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -0,0 +1,266 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	siteEditorStyleVariations: async (
+		{ admin, editor, page, pageUtils, requestUtils },
+		use
+	) => {
+		await use(
+			new SiteEditorStyleVariations( {
+				admin,
+				editor,
+				page,
+				pageUtils,
+				requestUtils,
+			} )
+		);
+	},
+} );
+
+test.describe( 'Global styles variations', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme(
+			'gutenberg-test-themes/style-variations'
+		);
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'should have three variations available with the first one being active', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await page.click( 'role=button[name="Styles"i]' );
+		await siteEditorStyleVariations.openOtherStyles();
+
+		const variations = page.locator(
+			'.edit-site-global-styles-variations_item'
+		);
+
+		await expect( variations ).toHaveCount( 3 );
+
+		// To avoid having to assert class names, https://github.com/WordPress/gutenberg/pull/41591
+		// adds aria-label and aria-current attributes to the style variation buttons.
+		// Once that PR is merged we can refactor this, e.g., await expect( variations.first() ).toHaveAttribute( 'aria-current', true );
+		await expect( await variations.first() ).toHaveClass( /is\-active/ );
+		await expect( await variations.nth( 1 ) ).not.toHaveClass(
+			/is\-active/
+		);
+		await expect( await variations.nth( 2 ) ).not.toHaveClass(
+			/is\-active/
+		);
+	} );
+
+	test( 'should apply preset colors and font sizes in a variation', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyPinkVariation();
+		await siteEditorStyleVariations.openPreviousGlobalStylesPanel();
+
+		await siteEditorStyleVariations.openColorsPanel();
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors background styles"] .component-color-indicator'
```

Comment:
It might be worth it to add `data-testid` to the `component-color-indicator` so that we don't have to use class names here. Could be in another PR though.